aws/resource_aws_glue_security_configuration.go

```diff
@@ -0,0 +1,342 @@
+		return fmt.Errorf("error reading Glue Security Configuration (%s): %s", d.Id(), err)
+	}
+
+	securityConfiguration := output.SecurityConfiguration
+	if securityConfiguration == nil {
+		log.Printf("[WARN] Glue Security Configuration (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	if err := d.Set("encryption_configuration", flattenGlueEncryptionConfiguration(securityConfiguration.EncryptionConfiguration)); err != nil {
+		return fmt.Errorf("error setting encryption_configuration: %s", err)
+	}
+
+	d.Set("name", securityConfiguration.Name)
+
+	return nil
+}
+
+func resourceAwsGlueSecurityConfigurationDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).glueconn
+
+	log.Printf("[DEBUG] Deleting Glue Security Configuration: %s", d.Id())
+	err := deleteGlueSecurityConfiguration(conn, d.Id())
+	if err != nil {
+		return fmt.Errorf("error deleting Glue Security Configuration (%s): %s", d.Id(), err)
+	}
+
+	return nil
+}
+
+func deleteGlueSecurityConfiguration(conn *glue.Glue, name string) error {
+	input := &glue.DeleteSecurityConfigurationInput{
+		Name: aws.String(name),
+	}
+
+	_, err := conn.DeleteSecurityConfiguration(input)
+
+	if isAWSErr(err, glue.ErrCodeEntityNotFoundException, "") {
+		return nil
+	}
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func expandGlueCloudWatchEncryption(l []interface{}) *glue.CloudWatchEncryption {
+	if len(l) == 0 || l[0] == nil {
+		return nil
+	}
+
+	m := l[0].(map[string]interface{})
+
+	cloudwatchEncryption := &glue.CloudWatchEncryption{
+		CloudWatchEncryptionMode: aws.String(m["cloudwatch_encryption_mode"].(string)),
+	}
+
+	if v, ok := m["kms_key_arn"]; ok {
+		cloudwatchEncryption.KmsKeyArn = aws.String(v.(string))
+	}
+
+	return cloudwatchEncryption
+}
+
+func expandGlueEncryptionConfiguration(l []interface{}) *glue.EncryptionConfiguration {
+	if len(l) == 0 || l[0] == nil {
+		return nil
+	}
+
+	m := l[0].(map[string]interface{})
+
+	encryptionConfiguration := &glue.EncryptionConfiguration{
+		CloudWatchEncryption:   expandGlueCloudWatchEncryption(m["cloudwatch_encryption"].([]interface{})),
+		JobBookmarksEncryption: expandGlueJobBookmarksEncryption(m["job_bookmarks_encryption"].([]interface{})),
+		S3Encryption:           expandGlueS3Encryptions(m["s3_encryption"].([]interface{})),
+	}
+
+	return encryptionConfiguration
+}
+
+func expandGlueJobBookmarksEncryption(l []interface{}) *glue.JobBookmarksEncryption {
+	if len(l) == 0 || l[0] == nil {
+		return nil
+	}
+
+	m := l[0].(map[string]interface{})
+
+	jobBookmarksEncryption := &glue.JobBookmarksEncryption{
+		JobBookmarksEncryptionMode: aws.String(m["job_bookmarks_encryption_mode"].(string)),
+	}
+
+	if v, ok := m["kms_key_arn"]; ok {
+		jobBookmarksEncryption.KmsKeyArn = aws.String(v.(string))
+	}
+
+	return jobBookmarksEncryption
+}
+
+func expandGlueS3Encryptions(l []interface{}) []*glue.S3Encryption {
+	s3Encryptions := make([]*glue.S3Encryption, len(l))
```

Comment:
the nil check below means we could have a different list length to items in the list; would suggest we default to 0

Reply:
Ah thank you for catching that, will update.